src/transformers/models/dpr/modeling_tf_dpr.py

```diff
@@ -144,12 +144,12 @@ class TFDPRReaderOutput(ModelOutput):
     attentions: Optional[Tuple[tf.Tensor]] = None
 
 
-class TFDPREncoder(TFPreTrainedModel):
+class TFDPREncoderLayer(tf.keras.layers.Layer):
 
     base_model_prefix = "bert_model"
 
-    def __init__(self, config: DPRConfig, *args, **kwargs):
-        super().__init__(config, *args, **kwargs)
+    def __init__(self, config: DPRConfig, **kwargs):
+        super().__init__(**kwargs)
```

Comment:
Can we also store the `config` here to avoid the gymnastic of looking in `self.bert_model` for it?

Reply:
Also, just below at line 156, the line
```
self.bert_model.config = config
```
should not be necessary anymore since `TFBertMainLayer` stores the config .

Then the assert on line 158 should just use `config`.